examples/common/python/connectors/ethereum/contracts/proxy-model/WorkOrderRegistry.sol

```diff
@@ -0,0 +1,171 @@
+/* Copyright 2019 iExec Blockchain Tech
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at 
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+pragma solidity ^0.5.0;
+
+import "./libs/SignatureVerifier.sol";
+
+contract WorkOrderRegistry is SignatureVerifier
+{
+	uint256 constant public TIMEOUT = 1 days;
+
+	enum WorkOrderStatus
+	{
+		NULL,
+		ACTIVE,
+		COMPLETED,
+		FAILED
+	}
+
+	struct WorkOrder
+	{
+		uint256 status;
+		uint256 timestamp;
+		bytes32 workerID;
+		bytes32 requesterID;
+		bytes   request;
+		address verifier;
+		uint256 returnCode;
+		bytes   response;
+	}
+
+	// workOrderID → workOrder
+	mapping(bytes32 => WorkOrder) m_workorders;
+
+	event workOrderSubmitted(
+		bytes32 indexed workOrderID,
+		bytes32 indexed workerID,
+		bytes32 indexed requesterID);
+
+	event workOrderCompleted(
+		bytes32 indexed workOrderID,
+		uint256 indexed workOrderReturnCode);
+
+	event workOrderTimedout(
+		bytes32 indexed workOrderID);
+
+	constructor()
+	public
+	{}
+
+	function workOrderSubmit(
+		bytes32       _workerID,
+		bytes32       _requesterID,
+		bytes memory _workOrderRequest,
+		address       _verifier,
+		bytes32       _salt)
+	public returns (
+		uint256       errorCode
+	) {
+		bytes32 workOrderID = keccak256(abi.encode(
+			_workerID,
+			_requesterID,
+			_workOrderRequest,
+			_verifier,
+			_salt
+		));
+
+		WorkOrder storage wo = m_workorders[workOrderID];
+		require(wo.status == uint256(WorkOrderStatus.NULL), "wo-already-submitted");
+		wo.status      = uint256(WorkOrderStatus.ACTIVE);
+		wo.timestamp   = now;
+		wo.workerID    = _workerID;
+		wo.requesterID = _requesterID;
+		wo.request     = _workOrderRequest;
+		wo.verifier    = _verifier;
+
+		emit workOrderSubmitted(
+			workOrderID,
+			_workerID,
+			_requesterID);
```

Comment:
I assume there are changes to spec. Publicly released spec mentions workOrderNew event is emitted by workOrderSubmit .
event workOrderNew (bytes32 indexed workOrderId,
        bytes32 indexed workerId,
        bytes32 indexed requesterId,
        string workOrderRequest,
        uint32 errorCode,
        address senderAddress,
        byte4 version)

Reply:
Yes, part of the spec change was about improving name consistency